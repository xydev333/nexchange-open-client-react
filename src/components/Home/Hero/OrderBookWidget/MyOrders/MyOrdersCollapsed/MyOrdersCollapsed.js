import React from 'react';
import { I18n } from 'react-i18next';
import arrow from 'Img/arrow-right-2.svg';
import { Link } from 'react-router-dom';

import styles from './MyOrdersCollapsed.scss';


const MyOrdersCollapsed = props => {
    const myOrdersList = <div className={styles.list}>
    <div className={styles.orders}>
      {props.myOrders.slice(0, 5).map((order) => {
        const isReverse = order.order_type === 0;
        const base = !isReverse ? 'base' : 'quote';
        const quote = !isReverse ? 'quote' : 'base';
        return (
          <Link to={`/order/${order.unique_reference}`} key={order.unique_reference} className={styles.order}>
            <div className={styles.coin}>
                <i className={`${styles.icon} coin-icon cc ${order.pair[base].code}`} />
                <span className={`${styles.code} hidden-xs hidden-ms hidden-sm`}>{order.pair[base].code}</span>
                <span className={styles.amount}>{parseFloat(order[`amount_${base}`]).toString().substring(0,9)}</span>
              </div>
              <img src={arrow} className={styles.arrow} alt="Arrow" />
              <div className={styles.coin}>
                <i className={`${styles.icon} coin-icon cc ${order.pair[quote].code}`} />
                <span className={`${styles.code} hidden-xs hidden-ms hidden-sm`}>{order.pair[quote].code}</span>
                <span className={styles.amount}>{parseFloat(order[`amount_${quote}`]).toString().substring(0,9)}</span>
              </div>
          </Link>);
      })}
    </div>
    <div className={`${styles.viewAll}`} 
         onClick={() => props.shouldRedirect ? props.goToMyOrders() : props.expandMyOrders()}>
      <a>View All My Orders</a>
    </div>
  </div>;
  return (
        <I18n ns="translations">
            {(t, i18n) => (
                    <div id='myOrders' className={`${styles.container}`}>
                        <div className={`${styles.heading}`}><h4>My Orders</h4></div>
                        <div className={`${styles.content}`}>
                        {_.isEmpty(props.myOrders) 
                            ? <p>No order history...</p>
                            : myOrdersList}
                        </div>
                    </div>
            )}
        </I18n>
  );
};

export default MyOrdersCollapsed;
