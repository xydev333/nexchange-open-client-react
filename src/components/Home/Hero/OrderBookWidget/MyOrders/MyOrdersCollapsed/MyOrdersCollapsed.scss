
.container {
    margin: 15px 5px;
    max-height: 400px;
    border: 1px solid lighten( $gray, 30);

    .heading {
        text-align: center;
        padding: 8px 0px;
        background-color: lighten( $gray, 30);
        h4 {
            margin: 0;
            font-size: 16px;
            font-weight: bold;
        }
    }

    .content {
        p {
            padding: 10px 10px 0px 10px;
        }
        .list {
            padding-top: 15px;
            padding: 10px 10px 0px 10px;

            .orders {
                max-height: 235px;
                overflow: auto;
                .order {
                    display: flex;
                    align-items: center;
                    padding: 8px 0px;
                    border-bottom: 1px solid lighten( $gray, 30);

                    .coin {
                        display: flex;
                        align-items: center;
                        flex: 1;

                        .icon {
                            flex: 1;
                            font-size: 20px;  
                        }
                        .code {
                            flex: 1;
                            font-size: 14px;  
                            font-weight: bold;
                            margin-left: 5px;
                        }
                        .amount {
                            flex: 4;
                            font-size: 14px;  
                            margin-left: 5px;
                        }
                    }
                    .arrow {
                        margin-right: 10px;
                        margin-left: 10px;
                        width: 16px;
                    }
                }
            }

            .viewAll {
                padding: 15px;
                text-align: center;
                a,&:hover {
                    color: $blue;
                    font-size: 14px;
                    font-weight: bold;
                    text-decoration: none;
                    text-transform: uppercase;
                    cursor: pointer;
                }
            }
        }
    }
}
