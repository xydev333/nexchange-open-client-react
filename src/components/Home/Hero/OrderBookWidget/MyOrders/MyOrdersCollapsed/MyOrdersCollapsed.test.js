import React from 'react';
import { shallow } from 'enzyme';
import MyOrdersCollapsed from './MyOrdersCollapsed';

describe('MyOrdersCollapsed', () => {
  let wrapShallow;

  beforeEach(() => {
    wrapShallow = shallow(<MyOrdersCollapsed myOrders={[]} expandMyOrders={jest.fn()} />).dive();
  });

  it('renders correctly', () => {
    expect(wrapShallow).toMatchSnapshot();
  });
});

