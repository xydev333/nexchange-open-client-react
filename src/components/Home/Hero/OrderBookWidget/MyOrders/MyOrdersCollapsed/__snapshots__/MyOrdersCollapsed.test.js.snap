// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrdersCollapsed renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
  >
    [Function]
  </I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="heading"
        >
          <h4>
            My Orders
          </h4>
        </div>,
        <div
          className="content"
        >
          <p>
            No order history...
          </p>
        </div>,
      ],
      "className": "container",
      "id": "myOrders",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            My Orders
          </h4>,
          "className": "heading",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Orders",
          },
          "ref": null,
          "rendered": "My Orders",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            No order history...
          </p>,
          "className": "content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No order history...",
          },
          "ref": null,
          "rendered": "No order history...",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="heading"
          >
            <h4>
              My Orders
            </h4>
          </div>,
          <div
            className="content"
          >
            <p>
              No order history...
            </p>
          </div>,
        ],
        "className": "container",
        "id": "myOrders",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              My Orders
            </h4>,
            "className": "heading",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Orders",
            },
            "ref": null,
            "rendered": "My Orders",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              No order history...
            </p>,
            "className": "content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No order history...",
            },
            "ref": null,
            "rendered": "No order history...",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
